Add Nuxt 3 migration project proposal

obulat opened the pull request:
<!-- prettier-ignore -->
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->

Part of #411

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->
Tagging @sarayourfriend and @zackkrida for their experience with frontend.

## Process
This discussion is following the Openverse decision-making process. Information
about this process can be found
[on the Openverse documentation site](https://docs.openverse.org/meta/decision_making/index.html).
Requested reviewers or participants will be following this process. If you are
being asked to give input on a specific detail, you do not need to familiarise
yourself with the process and follow it.

## Current round

This discussion is currently in the **Decision round**.

The deadline for review of this round is **2023-06-07**

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

sarayourfriend commented on documentation/projects/proposals/nuxt_3_migration/20230604-project_proposal_nuxt_3_migration.md:
```diff
@@ -0,0 +1,72 @@
+- All the current functionality should work as expected.
+- All the dependencies should be either updated to latest versions that support
+  Nuxt 3 or replaced with the ones that support Nuxt 3.
+- (Optional) Nuxt app should use node version 20, and pnpm version 8.
```
The `pnpm` version upgrade does seem optional to me, but moving off of Node 16 should either be a mandatory requirement of this, either to actually do it or identify the specific things that need to be done beyond Nuxt 3 upgrades (even in passing, to set up a subsequent implementation plan). Node 16 reaching EOL while we're still on it introduces a significant risk of security issues that we should prioritise avoiding.

obulat replied:
I'll update it to say:
- Nuxt app should migrate from node version 16 to the active LTS version 18 (or the current version 20, which will become active LTS on 2023-10-24.
- (Optional) Nuxt app should migrate to `pnpm` version 8

sarayourfriend reviewed (changes requested):
LGTM save the detail about the Node upgrade being optional. If we want to add wiggle room, we could say that the requirement should be that Node is upgradeable (not necessarily upgraded, if we need another implementation plan to cover dangling, non-Nuxt version related issues) to current LTS but not necessarily the _latest_ node.

The pnpm upgrade seems 100% optional to me and is just a nice to have. I'd be fine if it didn't happen and was handled in a separate smaller implementation plan if it has complexities to handle.

zackkrida commented:
I also think we can shortcut the rounds quickly. I agree with the changes to making the pnpm upgrade entirely optional / a separate project and defaulting to Node 18 as well.

obulat commented on documentation/projects/proposals/nuxt_3_migration/20230604-project_proposal_nuxt_3_migration.md:
```diff
@@ -0,0 +1,72 @@
+- All the current functionality should work as expected.
+- All the dependencies should be either updated to latest versions that support
+  Nuxt 3 or replaced with the ones that support Nuxt 3.
+- (Optional) Nuxt app should use node version 20, and pnpm version 8.
```
```suggestion

- Nuxt app should migrate from node version 16 to the active LTS version 18 (or the current version 20, which will become active LTS on 2023-10-24).
- (Optional) Nuxt app should migrate to pnpm version 8

```

obulat commented:
Based on the suggestions in the comments to short-circuit this discussion, I added the changes regarding the node and pnpm versions, and moved this discussion to the Decision step.

zackkrida commented on documentation/projects/proposals/nuxt_3_migration/20230604-project_proposal_nuxt_3_migration.md:
```diff
@@ -0,0 +1,75 @@
+- All the current functionality should work as expected.
+- All the dependencies should be either updated to latest versions that support
+  Nuxt 3 or replaced with the ones that support Nuxt 3.
+
+- Nuxt app should migrate from node version 16 to the active LTS version 18 (or the current version 20, which will become active LTS on 2023-10-24).
```
```suggestion
- Nuxt app should migrate from node version 16 to the active LTS version 18 (or the current version 20, which will become active LTS on 2023-10-24).
```

zackkrida reviewed (approved):
LGTM. I think this project would have been _okay_ without a project plan, and having one that is concise and ads further clarification about specific version numbers for Node is useful. LGTM.

sarayourfriend reviewed (approved):
LGTM. I also agree that this would have gone okay without a plan but I am glad we got to have the discussion about the Node version assumption and make it explicit. Thanks, Olga!

obulat commented on documentation/projects/proposals/nuxt_3_migration/20230604-project_proposal_nuxt_3_migration.md:
```diff
@@ -0,0 +1,75 @@
+
+<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->
+
+- [ ] @zackkrida - for previous experience of migrating the project to Nuxt
+- [ ] @sarayourfriend - for extensive experience on the frontend
```
```suggestion
- [x] @zackkrida - for previous experience of migrating the project to Nuxt
- [x] @sarayourfriend - for extensive experience on the frontend
```
